### PR TITLE
Add tz= to fread

### LINKED ISSUE #4600: Add tz= to fread()
Follow up to #4464 as discussed there.

### PR DESCRIPTION
**Pull request by mattdowle**
Closes #4600

### CODE REVIEW

File: R/fread.R [hunk 1]
```diff
@@ -267,8 +267,14 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir())
     if (is.integer(skip)) skip = skip + n_read
   }
   warnings2errors = getOption("warn") >= 2
+  stopifnot(identical(tz,"UTC") || identical(tz,""))
+  if (tz=="") {
+    tt = Sys.getenv("TZ", unset=NA_character_)
```
Comment (MichaelChirico) [inline R/fread.R:272]: From `?Sys.timezone`

> As from R 3.5.0, when a time zone location is first found in a session, its value is cached in object .sys.timezone in the base environment.

Should we check this as well? [1] check `.sys.timezone` [2] check `Sys.getenv('TZ')`.

Not sure it's worth getting that far into the weeds.

Comment (mattdowle) [reply R/fread.R:272]: All I've ever seen in use, especially production use, is the TZ environment variable. I think that's universal and standard across languages and OS, afaik. I don't even know what a "time zone location" is, but it sounds like something you'd need if we need to distinguish different local time zones, which we don't need. All we need is UTC or not-UTC. But anyway, we don't want the cached value at the start of the R session; if a user enters `Sys.setenv(TZ="UTC")`, as we now do in tests, then we want to recognize that latest value.

Comment (mattdowle) [reply R/fread.R:272]: And it comes back to the strangeness that if I do `Sys.setenv(TZ="UTC")`, `Sys.timezone()` isn't affected.  How then, do I change the value of `Sys.timezone()` if it isn't via the standard `TZ` envornment variable?!

Comment (MichaelChirico) [reply R/fread.R:272]: ok makes sense to me

Comment (MichaelChirico) [reply R/fread.R:272]: maybe related to the caching thing within session

**Comment by mattdowle** [comment]
Beats me why the Travis CI is failing. The log just seems to be chopped off as though it is being killed after 7 mins. Have tried rerunning a few times with no luck.

**Comment by MichaelChirico** [comment]
I see this under the `Checking package` heading

```
 test.data.table() running: /home/travis/build/Rdatatable/data.table/data.table.Rcheck/data.table/tests/tests.Rraw 
  Test 1743.25 ran without errors but failed check that x equals y:
  > x = fread("a,b,c\n2015-06-01 11:00:00,1,ae", colClasses = c("POSIXct",      "integer", "character")) 
                       a b  c  [Key= Types=dou,int,cha Classes=POS,int,cha]
  1: 2015-06-01 11:00:00 1 ae                                              
  > y = data.table(a = as.POSIXct("2015-06-01 11:00:00"), b = 1L, c = "ae") 
                       a b  c  [Key= Types=dou,int,cha Classes=POS,int,cha]
  1: 2015-06-01 11:00:00 1 ae                                              
  Column 'a': Attributes: < Component "tzone": 1 string mismatch >
  Test 1743.26 ran without errors but failed check that x equals y:
  > x = fread("a,b,c,d,e,f,g,h\n1,k,2015-06-01 11:00:00,a,1.5,M,9,0",      colClasses = list(POSIXct = "c", character = "b"), drop = c("a",          "b"), logical01 = TRUE) 
                       c d   e f g     h
  1: 2015-06-01 11:00:00 a 1.5 M 9 FALSE
      [Key= Types=dou,cha,dou,cha,int,log Classes=POS,cha,num,cha,int,log]
  1:                                                                      
  > y = ans <- data.table(c = as.POSIXct("2015-06-01 11:00:00"), d = "a",      e = 1.5, f = "M", g = 9L, h = FALSE) 
                       c d   e f g     h
  1: 2015-06-01 11:00:00 a 1.5 M 9 FALSE
      [Key= Types=dou,cha,dou,cha,int,log Classes=POS,cha,num,cha,int,log]
  1:                                                                      
  Column 'c': Attributes: < Component "tzone": 1 string mismatch >
  Test 1743.27 ran without errors but failed check that x equals y:
  > x = fread("a,b,c,d,e,f,g,h\n1,k,2015-06-01 11:00:00,a,1.5,M,9,0",      colClasses = list(POSIXct = "c", character = 2), drop = c("a",          "b"), logical01 = TRUE) 
                       c d   e f g     h
  1: 2015-06-01 11:00:00 a 1.5 M 9 FALSE
      [Key= Types=dou,cha,dou,cha,int,log Classes=POS,cha,num,cha,int,log]
  1:                                                                      
  > y = ans 
                       c d   e f g     h
  1: 2015-06-01 11:00:00 a 1.5 M 9 FALSE
      [Key= Types=dou,cha,dou,cha,int,log Classes=POS,cha,num,cha,int,log]
  1:                                                                      
  Column 'c': Attributes: < Component "tzone": 1 string mismatch >
  Test 1760 not run because this session either has no OpenMP or has been limited to one thread (e.g. under UBSAN and ASAN)
  Test 2150.13 ran without errors but failed check that x equals y:
  > x = fread("a,b\n2015-01-01,1.1\n2015-01-02 01:02:03,1.2") 
                       a   b  [Key= Types=dou,dou Classes=POS,num]
  1: 2015-01-01 00:00:00 1.1                                      
  2: 2015-01-02 01:02:03 1.2                                      
  > y = data.table(a = c("2015-01-01", "2015-01-02 01:02:03"), b = c(1.1,      1.2)) 
                       a   b  [Key= Types=cha,dou Classes=cha,num]
  1:          2015-01-01 1.1                                      
  2: 2015-01-02 01:02:03 1.2                                      
  Datasets have different column modes. First 3: a(numeric!=character)
  Test 2150.17 ran without errors but failed check that x equals y:
  > x = fread("a,b,c\n2015-01-01,2015-01-02,2015-01-03 01:02:03", colClasses = c("Date",      "IDate", "POSIXct")) 
              a          b                   c
  1: 2015-01-01 2015-01-02 2015-01-03 01:02:03
      [Key= Types=dou,int,dou Classes=Dat,IDa,POS]
  1:                                              
  > y = ans 
              a          b                   c
  1: 2015-01-01 2015-01-02 2015-01-03 01:02:03
      [Key= Types=dou,int,dou Classes=Dat,IDa,POS]
  1:                                              
  Column 'c': Attributes: < Component "tzone": 1 string mismatch >
  Test 2150.18 ran without errors but failed check that x equals y:
  > x = fread("a,b,c\n2015-01-01,2015-01-02,2015-01-03 01:02:03", colClasses = c("Date",      NA, NA)) 
              a          b                   c
  1: 2015-01-01 2015-01-02 2015-01-03 01:02:03
      [Key= Types=dou,int,dou Classes=Dat,IDa,POS]
  1:                                              
  > y = data.table(a = as.Date("2015-01-01"), b = as.IDate("2015-01-02"),      c = "2015-01-03 01:02:03") 
              a          b                   c
  1: 2015-01-01 2015-01-02 2015-01-03 01:02:03
      [Key= Types=dou,int,cha Classes=Dat,IDa,cha]
  1:                                              
  Datasets have different column modes. First 3: c(numeric!=character)
  
  Fri Jul 17 06:09:47 2020  endian==little, sizeof(long double)==16, sizeof(pointer)==8, TZ=='UTC', Sys.timezone()=='UTC', Sys.getlocale()=='LC_CTYPE=en_US.UTF-8;LC_NUMERIC=C;LC_TIME=en_US.UTF-8;LC_COLLATE=en_US.UTF-8;LC_MONETARY=en_US.UTF-8;LC_MESSAGES=en_US.UTF-8;LC_PAPER=en_US.UTF-8;LC_NAME=C;LC_ADDRESS=C;LC_TELEPHONE=C;LC_MEASUREMENT=en_US.UTF-8;LC_IDENTIFICATION=C', l10n_info()=='MBCS=TRUE; UTF-8=TRUE; Latin-1=FALSE', getDTthreads()=='omp_get_num_procs()==2; R_DATATABLE_NUM_PROCS_PERCENT==unset (default 50); R_DATATABLE_NUM_THREADS==unset; R_DATATABLE_THROTTLE==unset (default 1024); omp_get_thread_limit()==2147483647; omp_get_max_threads()==2; OMP_THREAD_LIMIT==unset; OMP_NUM_THREADS==unset; RestoreAfterFork==true; data.table is using 1 threads with throttle==1024. See ?setDTthreads.'
  Error in test.data.table() : 
    6 errors out of 9992. Search tests/tests.Rraw for test numbers: 1743.25, 1743.26, 1743.27, 2150.13, 2150.17, 2150.18.
```

**Comment by mattdowle** [comment]
Thanks! The section that fails is normally expanded in that view, I thought.  Anyway...

**Pull request merged**